### PR TITLE
N-gons and Texture coordinates in Rhino

### LINKED ISSUE #785: Rhino: Texture Coordinates on Meshes cause invalid meshes
See this discussion on discourse: https://speckle.community/t/blender-to-rhino-hangs-program/1876

As @JR-Morgan says, we might end up with a bigger discussion around texture coordinates encoding, as apparently Blender does it differently than Sketchup (and, yes, Rhino seems different too). Three cheers for interop!

### PR DESCRIPTION
**Pull request by JR-Morgan**
## Description

Rhino:
- Fixes #785 issue in rhino with texture coordinates causing failed stream
- Fixes issue where receiving n-gons leads to the entire stream failing
- Added support for N-gon ToSpeckle and ToNative. (see #780)

Objects:
- Added conveniance methods to `Mesh` and `Point`

## Type of change

- Bug fix and New feature (no breaking changes)

Several objects methods/constructors have been made obsolete. Still safe to use/call them, but they may be removed a few versions down the line. For each obsolete method, the new method/workaround is super clear, and most of the connectors have been updated to use them.

## How has this been tested?

Rhino:
- Texture coordinates - Manual test
    Current method of testing is to receive a stream with texture coordinates and apply a textured material.
- Mesh faces - Manual test
    Current method of testing is to recieve [N-gon test stream](https://latest.speckle.dev/streams/19f0ffa142/branches/main), send back to speckle, and observe JSON.
- [ ] TODO: Add n-gons to rhino test stream

Objects:
- Objects conveniance methods
   `Mesh.AlignVerticesWithTexCoordsByIndex` is tested through Unit tests in `Objects\Tests\Tests.csproj

## Docs

Texture coordiantes:
- No docs to be updated as of yet. Docs will be written as part of #797.

Ngons:
- [ ] TODO, update rhino docs mentioning support and limitations

### CODE REVIEW

**Comment by JR-Morgan** [comment]
## Pre PR faces: (only tris and quads with indecies of 0 and 1)
image

N-gons behaviour:
- ToNative:  If there is an n-gon anywhere in the stream, the entire stream will fail to receive
- ToSpeckle:  would be sent triangulated

## Post PR faces: (0-1, n)
image
Note : Discrepancies in concave and complex n-gons appear to be a limitation of Rhino

N-gons behaviour:
- ToNative: as N-gons
- ToSpeckle: as N-gons

File: Objects/Objects/Geometry/Mesh.cs [hunk 1]
```diff
@@ -38,14 +40,115 @@ public Mesh()
...
+    
+    public int VerticesCount => vertices.Count / 3;
+    public int TextureCoordinatesCount => textureCoordinates.Count / 2;
+
+    /// <summary>
+    /// Gets a vertex as a <see cref="Point"/> by <paramref name="index"/>
+    /// </summary>
+    /// <param name="index">The index of the vertex</param>
+    /// <returns>Vertex as a <see cref="Point"/></returns>
+    public Point GetPointAtIndex(int index)
```
Comment (clairekuang) [inline Objects/Objects/Geometry/Mesh.cs:74]: Rename to `GetPoint` to align with additional `GetPoints` convenience method

File: Objects/Objects/Geometry/Mesh.cs [hunk 2]
```diff
@@ -38,14 +40,115 @@ public Mesh()
...
+      applicationId,
+      units
+    )
+    { }
+
+    #region Convenience Methods
+    
+    public int VerticesCount => vertices.Count / 3;
+    public int TextureCoordinatesCount => textureCoordinates.Count / 2;
+
```
Comment (clairekuang) [inline Objects/Objects/Geometry/Mesh.cs:68]: Add `GetPoints()` convenience method as in `Curve` class: https://github.com/specklesystems/speckle-sharp/blob/7fba43413cc7e2edba2d88779d94901ffb838cfa/Objects/Objects/Geometry/Curve.cs#L63

**Review by clairekuang — COMMENTED** [review]
Didn't thoroughly test in the rhino connector but I trust you! Just added comments to the `Mesh` convenience methods to better align with other classes

**Review by clairekuang — APPROVED** [review]

**Pull request merged**